kufar/books/main.go

```diff
@@ -208,10 +209,22 @@ func (d Database) SignInUser(w http.ResponseWriter, r *http.Request) {
 
 	cred.Username = credReq.Username
 
-	session.Values["user"] = cred.UUID
-	if err = session.Save(r, w); err != nil {
-		http.Error(w, http.StatusText(http.StatusInternalServerError)+": could not assign cookie", http.StatusInternalServerError)
-		return
+	credentials := r.Header.Get("Authorization")
+
+	switch {
+	case credentials != "":
+		r.SetBasicAuth(cred.Username, cred.Password)
```

Comment:
It looks a bit strange, but I've got your idea. It's not so easy to understand so we will think about some refactoring tomorrow.